src/framework-ondata.js

```diff
@@ -1,18 +1,164 @@
 /**
  * File: framework-ondata.js 
  */
+var _ = require('underscore');
+var util = require('util');
+
+var stateFactory = require('./create-state');
+var createLm = require('./create-lm');
+var vniManager = require('./vni-manager');
 
 /** 
- * This is a stub file to allow for a smaller PR and merge.  It will be replaced
- * with the real version of the code once the pipeline-component-factory is finalized.
- *
+ * The standard RDF pipeline framework ondata method. This is called whenever  
+ * there is new data on any port.  This method will update the input state,
+ * and determine if the component updater should be executed now or not. 
+ * If the updater should be executed, the fRunUpdater wrapper API will be
+ * called to to handle that.  The updater output (or error) will be stored 
+ * in output or error state, and those results will be sent on to any 
+ * down stream components for their processing.
+ * 
  * @this port context
- *
- * @param payload data payload on the input port
- * @param socketIndex (optional) this will be set only for ports configured for
- *        multiple inputs (multi/noflo addressable).
+ * @param payload data coming into the port
+ * @param socketIndex socket index if the port is a multiple input (addressable) 
  */
 module.exports = function( payload, socketIndex ) {
 
-     return;
+     var portName = this.name;
+
+     var vnid = payload.vnid || '';
+     var vni = this.nodeInstance.vni(vnid);
+
+     // Update the input state
+     var inputState = (payload.vnid) ? payload : stateFactory( vnid, payload );
+     vni.inputStates( portName, socketIndex, inputState );
+
+     // check if should run updater & run if necessary
+     if ( shouldRunUpdater( vni ) ) { 
+          
+        // Save vars we will need in the promise where the context is different
+        var outputPorts = this.nodeInstance.outPorts;
+
+        if ( _.isUndefined( this.nodeInstance.wrapper.fRunUpdater ) || 
+             ! _.isFunction( this.nodeInstance.wrapper.fRunUpdater ) ) { 
```

Comment:
Come to think of it, let's separate this into two different cases, with both cases allowing the pipeline to continue execution:

1. If there is no updater then write a warning to the error state (and log it to the console) just as if an updater had run and set error.data to that warning and updated error.lm .  

2. If there is an fRunUpdater but it is not a function, then write an error message to the error state (setting error.data and error.lm) and also log it to the console.

If we take this approach then I think it will be friendlier to users, because the misconfiguration of one node will not stop the whole pipeline from running.